main.py

```diff
@@ -1990,57 +1994,192 @@ def initialize_triggers(self):
+                parameter = None
+                parameter_label = None
+                parameter_documentation = None
+        else:
+            signature = None
+            signature_label = None
+            signature_documentation = None
+            parameter = None
+            parameter_label = None
+            parameter_documentation = None
+
+        content = io.StringIO()
+
+        content.write(r'''
+<html>
+    <body id="lsp">
+        <style>
+            body {
+                margin: 4px;
+            }
+            p {
+                margin: 0.1rem;
+            }
+            div.signature_block {
+                font-size: 0.95rem;
+                display: block;
+            }
+            span.number_of_signatures {
+                font-weight: bold;
+            }
+            span.active_signature {
+            }
+            span.active_parameter {
+                font-weight: bold;
+                text-decoration: underline;
+            }
+            div.signature_documentation {
+                display: block;
+                font-family: Helvetica;
+                font-size: 1.05rem;
+                margin: 0.5rem;
+            }
+            div.parameter_documentation {
+                display: block;
+                font-family: Helvetica;
+                font-size: 1.05rem;
+                margin: 0.5rem;
+            }
+        </style>
+''')
+
+        # Put the active signature in a signature_block class.
+        content.write('<div class="signature_block">')
+
+        # Write the active signature number and the total number of signatures.
+        if len(self.signatures) > 1 and self.active_signature in range(0, len(self.signatures)):
+            content.write('<span class="number_of_signatures">')
+            content.write(str(self.active_signature + 1))
+            content.write("/")
+            content.write(str(len(self.signatures)))
+            content.write("</span>\n")  # number_of_signatures
+
+        # Write the active signature and give special treatment to the active parameter (if found).
+        if signature_label:
+            content.write('<span class="active_signature">')
+            if parameter_label:
+                index = signature_label.find(parameter_label)
+            else:
+                index = -1
+            if index == -1:
+                content.write(signature_label)
             else:
-                # TODO: this hides too soon.
-                if self.view.is_popup_visible():
-                    self.view.hide_popup()
+                index_end = index + len(parameter_label)
+                content.write(signature_label[:index])
+                content.write('<span class="active_parameter">')
+                content.write(signature_label[index:index_end])
+                content.write("</span>")  # active_parameter
+                content.write(signature_label[index_end:])
+            content.write("</span>")  # active_signature
```

Comment:
Do I understand correctly that his code does the following?
```python
signature_label.replace(parameter_label, '<span class="active_parameter">{}</span>'.format(parameter_label))
```
I would really prefer the above style unless there is a good reason against it (yes, it's not handling all the cases, but that can be fixed).
That said, I wonder if it wouldn't be safer to build the signature manually using `parameters`? This looks like it might run into problems if someone names their arguments `(instance, i)`

Reply:
Ah yes, your line is much more elegant than my clumsy StringIO approach. I hope to have addressed all your issues tomorrow evening!